x/tally/keeper/gas_calculation.go

```diff
@@ -0,0 +1,222 @@
+package keeper
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/x/tally/types"
+)
+
+// GasCalculation stores the results of the canonical gas consumption calculations.
+type GasCalculation struct {
+	Burn          uint64 // filter and tally gas to be burned
+	Proxies       []ProxyGasUsed
+	Executors     []ExecutorGasUsed
+	ReducedPayout bool
+}
+
+type ProxyGasUsed struct {
+	PublicKey []byte
+	Amount    math.Int
+}
+
+type ExecutorGasUsed struct {
+	PublicKey string
+	Amount    math.Int
+}
+
+// DistributionsFromGasCalculation constructs a list of distribution messages
+// to be sent to the core contract based on the given gas calculation.
+func (k Keeper) DistributionsFromGasCalculation(ctx sdk.Context, reqID string, gasCalc GasCalculation, gasPrice math.Int, burnRatio math.LegacyDec) []types.Distribution {
+	dists := []types.Distribution{}
+	attrs := []sdk.Attribute{
+		sdk.NewAttribute(types.AttributeDataRequestID, reqID),
```

Comment:
I also need the blockheight at which the DR was issued to be able to uniquely identify which DR this gas result belongs to 😬